src/ast/diagnostic.rs

```diff
@@ -198,6 +198,43 @@ impl Span {
             variant,
         }
     }
+
+    pub fn to_lsp_range(self, files: &Files) -> Option<lsp_types::Range> {
+        let file = files.get(self.file).unwrap();
+        let char_span = file.char_span(self);
+
+        let mut start_line = 0;
+        let mut start_offset = 0;
+        let mut end_line = 0;
+        let mut end_offset = 0;
+
+        for (i, c) in file.source.chars().enumerate() {
```

Comment:
@umutdural Can you say why you need to iterate over the source code here again?

Reply:
char_span gives the span in terms of characters from the beginning of the file. However VSCode API requires spans (ranges) in line and offset format. I needed to find which start and end lines the char span corresponds to, thus I needed to count the number of newline characters until the beginning and end of the char span.

Reply:
Ah I see. This functionality is available in ariadne already. If you use a modified version of [this code](https://github.com/moves-rwth/caesar/blob/cd06d796a41477e8877cda4744460aba07720b2a/src/ast/diagnostic.rs#L127-L137), you should be able to obtain this data as well. Main function is [ariadne's `get_offset_line`](https://docs.rs/ariadne/latest/ariadne/struct.Source.html#method.get_offset_line).